builtin/logical/database/secret_creds.go

```diff
@@ -0,0 +1,138 @@
+package database
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+const SecretCredsType = "creds"
+
+func secretCreds(b *databaseBackend) *framework.Secret {
+	return &framework.Secret{
+		Type: SecretCredsType,
+		Fields: map[string]*framework.FieldSchema{
+			"username": &framework.FieldSchema{
```

Comment:
True story, I think the fields here aren't used, like, anywhere in Vault. I believe you can pull them out and just have the type/renew/revoke bits.